File: Infinity.Bnois.Web/Infinity.Bnois.Web/Infinity.Bnois.Web/app/views/daily-process/dailyProcessesController.js

(function () {

    'use strict';
    var controllerId = 'dailyProcessesController';
    angular.module('app').controller(controllerId, dailyProcessesController);
    dailyProcessesController.$inject = ['$state', 'dailyProcessService', 'notificationService', '$location'];

    function dailyProcessesController($state, dailyProcessService, notificationService, location) {
        /* jshint validthis:true */
        var vm = this;
        vm.dateModel = {};
        vm.promotionBoardId = 0;
        vm.promotionBoards = [];
        vm.promotionExecutionDate = null;
        vm.executePromotion = executePromotion;
        vm.executePromotionWithoutBoard = executePromotionWithoutBoard;
        vm.executePunishment = executePunishment;
        vm.executeSeniority = executeSeniority;
        vm.executeTransfer = executeTransfer;
        vm.executeAdvanceSearch = executeAdvanceSearch;
        vm.executeTransferZoneService = executeTransferZoneService;
        vm.executeNamingConvention = executeNamingConvention;
        vm.updateAgeServicePolicy = updateAgeServicePolicy;
        vm.uploadImageToFolder = uploadImageToFolder;
        vm.executeDatabaseBackup = executeDatabaseBackup;
        init();
        function init() {
            dailyProcessService.getDailyProcesses().then(function (data) {
                vm.dateModel.executionDate = new Date();
                    vm.promotionBoards = data.result;
                },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }

        function executePromotion() {
            dailyProcessService.executePromotion(vm.promotionBoardId).then(function (data) {
                notificationService.displaySuccess("Promotion Executed Successfully!");
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        function executePromotionWithoutBoard() {
            dailyProcessService.executePromotionWithoutBoard().then(function (data) {
                notificationService.displaySuccess("Promotion Without Board Executed Successfully!");
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        function executePunishment() {
            dailyProcessService.executePunishment().then(function (data) {
                notificationService.displaySuccess("Punishment Executed Successfully!");
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }

        function executeSeniority() {
            dailyProcessService.executeSeniority().then(function (data) {
                notificationService.displaySuccess("Seniority Executed Successfully!");
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }

        function executeTransfer() {
            dailyProcessService.executeTransfer().then(function (data) {
                notificationService.displaySuccess("Transfer Executed Successfully!");
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        function executeAdvanceSearch() {
            dailyProcessService.executeAdvanceSearch().then(function (data) {
              notificationService.displaySuccess("Advance Search Executed Successfully!");

            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        function executeTransferZoneService() {
            dailyProcessService.executeTransferZoneService().then(function (data) {
              notificationService.displaySuccess("Transfer Zone Service Executed Successfully!");

            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        function executeNamingConvention() {
            dailyProcessService.executeNamingConvention().then(function (data) {
                notificationService.displaySuccess("Naming Convention Executed Successfully!");
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }


        function updateAgeServicePolicy() {
            dailyProcessService.updateAgeServicePolicy().then(function (data) {
                notificationService.displaySuccess(data.result);
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        function executeDatabaseBackup() {
            dailyProcessService.executeDataBaseBackup().then(function (data) {
                notificationService.displaySuccess("Database Backup Successfully!");
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        function uploadImageToFolder() {
            dailyProcessService.uploadImageToFolder().then(function (data) {
                notificationService.displaySuccess(data.result);
            },
                function (errorMessage) {
                    notificationService.displayError(errorMessage.message);
                });
        }
        
    }

})();

File: Infinity.Bnois/Infinity.Bnois.Api/Controllers/DailyProcessesController.cs
using Infinity.Bnois.Api.Core;
using Infinity.Bnois.Api.Right;
using Infinity.Bnois.ApplicationService.Interface;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web.Http;
using System.Web.Http.Cors;

namespace Infinity.Bnois.Api.Controllers
{
    [RoutePrefix(BnoisRoutePrefix.DailyProcesses)]
    [EnableCors("*", "*", "*")]
    [ActionAuthorize(Feature = MASTER_SETUP.DAILY_PROCESSES)]
    public class DailyProcessesController : BaseController
    {
        private readonly IPromotionNominationService promotionNominationService;
        private readonly IPunishmentAccidentService punishmentAccidentService;
        private readonly IEmployeeService employeeService;
        private readonly IPhotoService photoService;
        private readonly ITransferService transferService;
        private readonly IPromotionBoardService promotionBoardService;
        private readonly IAdvanceSearchService advanceSearchService;
        private readonly IDashboardService dashboardService;
        public DailyProcessesController(IPromotionBoardService promotionBoardService,IPhotoService photoService,IEmployeeService employeeService,IPromotionNominationService promotionNominationService, 
            IPunishmentAccidentService punishmentAccidentService,ITransferService transferService, IAdvanceSearchService advanceSearchService, IDashboardService dashboardService)
        {
            this.promotionNominationService = promotionNominationService;
            this.punishmentAccidentService = punishmentAccidentService;
            this.employeeService = employeeService;
            this.transferService = transferService;
            this.photoService = photoService;
            this.promotionBoardService = promotionBoardService;
            this.advanceSearchService = advanceSearchService;
            this.dashboardService = dashboardService;
        }

        [HttpGet]
        [Route("get-daily-processes")]
        public async Task<IHttpActionResult> GetDailyProcesses()
        {
            List<SelectModel> promotionBoards = promotionBoardService.GetDailyProcesPromotionBoardSelectModels();
            return Ok(new ResponseMessage<List<SelectModel>>
            {
                Result = promotionBoards
            });
        }

        [HttpPost]
        [ModelValidation]
        [Route("execute-promotion")]
        public async Task<IHttpActionResult> ExecutePromotion(int promotionBoardId)
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = await promotionNominationService.ExecutePromotion(promotionBoardId)
            });
        }

        [HttpPost]
        [ModelValidation]
        [Route("execute-data-base-backup")]
        public async Task<IHttpActionResult> ExecuteDataBaseBackup()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = await promotionNominationService.ExecuteDatabaseBackup()
            });
        }


        [HttpPost]
        [ModelValidation]
        [Route("execute-promotion-without-board")]
        public async Task<IHttpActionResult> ExecutePromotionWithOutBoard()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = await promotionNominationService.ExecutePromotionWithOutBoard()
            });
        }


        [HttpPost]
        [ModelValidation]
        [Route("execute-transfer")]
        public  IHttpActionResult ExecuteTransfer()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result =  transferService.ExecuteTransfer()
            });
        }



        [HttpPost]
        [ModelValidation]
        [Route("execute-advance-search")]
        public  IHttpActionResult ExecuteAdvanceSearch()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result =  advanceSearchService.ExecuteAdvanceSearch()
            });
        }
        

        [HttpPost]
        [ModelValidation]
        [Route("execute-transfer-zone-service")]
        public  IHttpActionResult ExecuteTransferZoneService()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result =  advanceSearchService.ExecuteTransferZoneService()
            });
        }
        

        [HttpPost]
        [ModelValidation]
        [Route("update-sea-service")]
        public IHttpActionResult UpdateSeaService()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = advanceSearchService.UpdateSeaService()
            });
        }

        [HttpPost]
        [ModelValidation]
        [Route("update-sea-cmd-service")]
        public IHttpActionResult UpdateSeaCmdService()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = advanceSearchService.UpdateSeaCmdService()
            });
        }


        [HttpPost]
        [ModelValidation]
        [Route("update-sea-service-days")]
        public IHttpActionResult UpdateSeaServiceDays()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = advanceSearchService.UpdateSeaServiceDays()
            });
        }

        [HttpPost]
        [ModelValidation]
        [Route("update-sea-service-years")]
        public IHttpActionResult UpdateSeaServiceYears()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = advanceSearchService.UpdateSeaServiceYears()
            });
        }



        [HttpPost]
        [ModelValidation]
        [Route("execute-naming-convention")]
        public IHttpActionResult ExecuteNamingConvention()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = employeeService.ExecuteNamingConvention()
            });
        }



        [HttpGet]
        [Route("get-update-punishment")]
        public async Task<IHttpActionResult> GetPunishmentUpdate()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = await punishmentAccidentService.ExecutePunishmentProcess()
            });
        }

        [HttpGet]
        [Route("get-execute-seniority")]
        public async Task<IHttpActionResult> GetSeniorityUpdate()
        {
            return Ok(new ResponseMessage<bool>
            {
                Result = await employeeService.ExecuteSeniorityProcess()
            });
        }

        [HttpPut]
        [Route("update-age-service-policy")]
        public async Task<IHttpActionResult> UpdateAgeServicePolicy()
        {
            return Ok(new ResponseMessage<string>
            {
                Result = await employeeService.UpdateAgeServicePolicy()
            });
        }


        [HttpPost]
        [Route("upload-image-into-folder")]
        public async Task<IHttpActionResult> UploadImageToFolder()
        {
            return Ok(new ResponseMessage<string>
            {
                Result = await photoService.UploadImageToFolder()
            });
        }


    }
}
